lib/commands/crafter/add_office.rb

```diff
@@ -20,21 +19,15 @@ def applies_to?(request)
 
     def prepare(data)
       @office = Office.new(parse_office_location(data))
-      @crafter = User.profile(data[:user_id])
-      @apprentice = Apprentice.profile(data[:user_id])
+      @user = User.profile(data[:user_id])
     end
 
     def run
-      return ""                 unless user
+      return ""                 unless @user
       return office_unavailable unless @office.available?
 
-      if @crafter
-        @crafter.add_office(@office.location)
-        log_office_added
-      end
-
-      if @apprentice
-        @apprentice.add_office(@office.location)
+      if @user
+        @user.add_office(@office.location)
```

Comment:
if you add the user as an `attr_reader` you will not need to refactor the different variable that use `user`
For the very same reason I would also add `@office` as an `attr_reader` as it leave us some flexibility when required to refactor (here it's a great example)

Reply:
Yeah I agree. And we should probably also replace the `prepare` methods with `initialize`. Thoughts?